src/System.Collections/src/System/Collections/Generic/List.cs

```diff
@@ -740,7 +737,7 @@ public void Insert(int index, T item)
             // Note that insertions at the end are legal.
             if ((uint)index > (uint)_size)
             {
-                throw new ArgumentOutOfRangeException("index", SR.ArgumentOutOfRange_ListInsert);
+                throw new ArgumentOutOfRangeException("index", index, SR.ArgumentOutOfRange_Index);
```

Comment:
This doesn't look like a valid change.  ArgumentOutOfRange_Index talks about the index being less than the size of the collection, but here it can be equal to the size of the collection.  I'd leave this one as it was.